soc/st/stm32/stm32wbax/hci_if/linklayer_plat.c

```diff
@@ -314,3 +321,27 @@ void LINKLAYER_PLAT_SCHLDR_TIMING_UPDATE_NOT(Evnt_timing_t *p_evnt_timing) {}
 void LINKLAYER_PLAT_EnableOSContextSwitch(void) {}
 
 void LINKLAYER_PLAT_DisableOSContextSwitch(void) {}
+
+void LINKLAYER_PLAT_NotifyWFIEnter(void)
+{
+	/* Check if Radio state will allow the AHB5 clock to be cut */
+
+	/* AHB5 clock will be cut in the following cases:
+	 * - 2.4GHz radio is not in ACTIVE mode (in SLEEP or DEEPSLEEP mode).
+	 * - RADIOSMEN and STRADIOCLKON bits are at 0.
+	 */
+	if ((LL_PWR_GetRadioMode() != LL_PWR_RADIO_ACTIVE_MODE) ||
+	   ((__HAL_RCC_RADIO_IS_CLK_SLEEP_ENABLED() == 0) &&
+	   (LL_RCC_RADIO_IsEnabledSleepTimerClock() == 0))) {
+		AHB5_SwitchedOff = 1;
+	}
+}
```

Comment:
This is a STM32Cube function implemented using STM32Cube APIs, it should exist in STM32Cube package and matching file should be included as part of https://github.com/zephyrproject-rtos/hal_stm32/tree/main/lib/stm32wba/hci.

This should STM32Cube specific code should be maintained outside of Zephyr and forked.
Risk of including such code is that maintenance will be complex as we can't get code evolutions.
Also, it raises License questions.

Same for functions introduced in soc/st/stm32/stm32wbax/hci_if/ll_sys_if.c.

Reply:
A better separation has been done to move STM32CUBE API based code inside  https://github.com/zephyrproject-rtos/hal_stm32/tree/main/lib/stm32wba/hci